lib/consumer_supervisor.ex

```diff
@@ -262,6 +256,24 @@ defmodule ConsumerSupervisor do
     call(supervisor, :count_children)
   end
 
+  @doc """
+  Receives a template to initialize and a set of options.
+
+  This is typically invoked at the end of the init/1 callback of module-based supervisors.
+
+  This function returns a the child specification and the supervisor flags.
+
+  """
+  def init({_, _, _, _, _, _} = template, opts) do
+    {:ok, [template], opts}
+  end
+  if function_exported? Supervisor, :init, 2 do
```

Comment:
Yes, in places where it does not exist, it makes no sense (in my opinion) to even define this funtion implementation. If there is no supervisor.init, only a tuple should be allowed.